### PR TITLE
ite: drivers/i2c: it8xxx2: Don't check bus busy during repeat start

### PR DESCRIPTION
**Pull request by GTLin08**
If the transaction of write to read is divided into two transfers,
the second transfer will go to check bus busy and cause i2c reset.
This change adds flag to eliminate this situation.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

### CODE REVIEW

File: drivers/i2c/i2c_ite_it8xxx2.c [hunk 1]
```diff
@@ -732,20 +732,23 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
...
-	/* Make sure we're in a good state to start */
-	if (i2c_bus_not_available(dev)) {
-		/* reset i2c port */
-		i2c_reset(dev, I2C_RC_NO_IDLE_FOR_START);
-		/*
-		 * After resetting I2C bus, if I2C bus is not available
-		 * (No external pull-up), drop the transaction.
-		 */
+
+	if (data->i2ccs == I2C_CH_NORMAL) {
```
Comment (brockus-zephyr) [inline drivers/i2c/i2c_ite_it8xxx2.c:736]: Can you add a short comment indicating this condition is checking for restart

**Review by brockus-zephyr — COMMENTED** [review]
Just have the one readability comment requested. Otherwise, I will approve

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 1 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c:741]: Done

**Comment by GTLin08** [comment]
Any comments about this PR?

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 1 above)
Comment (brockus-zephyr) [reply drivers/i2c/i2c_ite_it8xxx2.c:741]: /*
	 * If the transaction of write to read is divided into two
	 * transfers, the repeat start transfer uses this flag to
	 * exclude checking bus busy.
	 */

**Review by brockus-zephyr — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]
Looks good with the comment change ("divided into twice" -> "divided into two") suggested by Denis.

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 1 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c:741]: Done

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**